src/main/java/core/basesyntax/strategy/SupplyHandler.java

```diff
@@ -0,0 +1,12 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class SupplyHandler implements OperationHendler {
+    @Override
+    public void operateTransaction(FruitTransaction transaction) {
+        int oldQuantity = Storage.fruitStorage.getOrDefault(transaction.getFruit(), 0);
```

Comment:
Extract `fruitStorage.getOrDefault(transaction.getFruit(), 0);` to the new method `Storage.get(String fruit)` in class Storage. For better compliance to Single Responsibility principe.